### PR TITLE
feature/v2-updates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR

**What change(s) does this PR introduce?** 

## 🚨 Breaking Changes 🚨
The `pin_promotion_report_pass_through_metric` variable has been renamed to `pinterest__pin_promotion_report_passthrough_metrics`.
## 🎉 Feature Enhancements 🎉
This PR includes the following changes:
- Addition of the following staging models which pull from the source counterparts. The inclusion of the additional `_report` source tables is to generate a more accurate representation of the Pinterest Ads data. For example, not all Ad types are included within the `pin_promotion_report` table. Therefore, the addition of the further grain reports will allow for more flexibility and accurate Pinterest Ad reporting. 
  - `stg_pinterest_ads__ad_group_report`
  - `stg_pinterest_ads__advertiser_report`
  - `stg_pinterest_ads__campaign_report`
  - `stg_pinterest_ads__keyword_report`
  - `stg_pinterest_ads__advertiser_history`
  - `stg_pinterest_ads__keyword_history`

- Inclusion of additional passthrough metrics: 
  - `pinterest__ad_group_report_passthrough_metrics`
  - `pinterest__campaign_report_passthrough_metrics`
  - `pinterest__advertiser_report_passthrough_metrics`
  - `pinterest__keyword_report_passthrough_metrics`

- README updates for easier navigation and use of the package. 
- Addition of identifier variables for each of the source tables to allow for further flexibility in source table direction within the dbt project.
- Included grain uniqueness tests for each staging table. 

**Did you update the CHANGELOG?** 

- [X] Yes

**Does this PR introduce a breaking change?**

- [X] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [X] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [X] Yes, This will actually resolve PR #15 as well
- [X] No 

**How did you test the PR changes?** 

 

- [X] CircleCi  
- [X] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [X] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Review by fivetran-sheringuyen — COMMENTED** [review]
Hey @fivetran-joemarkiewicz, great work on the PR. I left some in-line comments I'd like for you to check out, let me know if you have any questions.

This looks like it also takes care of PR #15 and allows us to remove the `feature/spark` branch for this repo! Additionally, did you still want to file feature requests for a video related model in the future?

File: models/src_pinterest_ads.yml [hunk 1]
```diff
@@ -87,4 +100,170 @@ sources:
...
+            description: Bid amount set for the keyword.
+          - name: campaign_id
+            description: "{{ doc('campaign_id') }}"
+          - name: match_type
+            description: Type of match the keyword is tied to. Either Exact or Broad.
+          - name: parent_type
+            description: Identifier of what grain the parent type is. Ad group or campaign.
+
+      - name: keyword_report
+        description: source table
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_pinterest_ads.yml:239]: Wow missed these descriptions 🤦. Thanks for catching this. I will update them!

File: models/src_pinterest_ads.yml [hunk 2]
```diff
@@ -87,4 +100,170 @@ sources:
           - name: end_time
             description: Ad group end time. 
           - name: _fivetran_synced
-            description: The time Fivetran synced a record.
\ No newline at end of file
+            description: "{{ doc('_fivetran_synced') }}"
+
+      - name: ad_group_report
+        description: source table
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_pinterest_ads.yml:106]: Updated.

File: models/src_pinterest_ads.yml [hunk 3]
```diff
@@ -87,4 +100,170 @@ sources:
...
+            description: "{{ doc('impression_2') }}"
+          - name: clickthrough_1
+            description: "{{ doc('clickthrough_1') }}"
+          - name: clickthrough_2
+            description: "{{ doc('clickthrough_2') }}"
+          - name: spend_in_micro_dollar
+            description: "{{ doc('spend_in_micro_dollar') }}"
+
+      - name: advertiser_history
+        description: source table
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_pinterest_ads.yml:135]: Updated.

File: models/src_pinterest_ads.yml [hunk 4]
```diff
@@ -87,4 +100,170 @@ sources:
...
+            description: Unique ID of the merchant associated with the advertiser.
+          - name: owner_user_id
+            description: Unique identifier of the owner user.
+          - name: updated_time
+            description: "{{ doc('updated_at') }}"
+          - name: _fivetran_synced
+            description: "{{ doc('_fivetran_synced') }}"
+
+      - name: advertiser_report
+        description: source table
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_pinterest_ads.yml:164]: Updated.

File: models/src_pinterest_ads.yml [hunk 5]
```diff
@@ -87,4 +100,170 @@ sources:
...
+            description: "{{ doc('impression_2') }}"
+          - name: clickthrough_1
+            description: "{{ doc('clickthrough_1') }}"
+          - name: clickthrough_2
+            description: "{{ doc('clickthrough_2') }}"
+          - name: spend_in_micro_dollar
+            description: "{{ doc('spend_in_micro_dollar') }}"
+
+      - name: campaign_report
+        description: source table
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_pinterest_ads.yml:185]: Updated.

File: models/src_pinterest_ads.yml [hunk 6]
```diff
@@ -87,4 +100,170 @@ sources:
...
+            description: "{{ doc('impression_2') }}"
+          - name: clickthrough_1
+            description: "{{ doc('clickthrough_1') }}"
+          - name: clickthrough_2
+            description: "{{ doc('clickthrough_2') }}"
+          - name: spend_in_micro_dollar
+            description: "{{ doc('spend_in_micro_dollar') }}"
+
+      - name: keyword_history
+        description: source table
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_pinterest_ads.yml:212]: Updated.

File: models/src_pinterest_ads.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_pinterest_ads.yml:239]: Updated.

File: models/docs.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+{% docs pin_promotion_id %} The ID of the related Pin promotion. {% enddocs %}
+
+{% docs campaign_id %} The ID of the related Campaign. {% enddocs %}
+
+{% docs advertiser_id %} The ID of the related Advertiser. {% enddocs %}
+
+{% docs impressions %} The number of impressions that occurred on the day of the record. {% enddocs %}
+
+{% docs clicks %} The number of clicks that occurred on the day of the record. {% enddocs %}
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/docs.md]: Ditto! Updated.

**Review by fivetran-sheringuyen — APPROVED** [review]
Looks good, thanks for the updates @fivetran-joemarkiewicz !

**Pull request merged**